### PR TITLE
Update gdp datasets

### PR DESCRIPTION
**Pull request by philippemiron**

### CODE REVIEW

**Review by selipot — APPROVED** [review]

**Comment by selipot** [comment]
Does the adapter function also need to be updated?

**Comment by philippemiron** [comment]
Looks like it.. but the dates do not match on the folder, https://www.aoml.noaa.gov/ftp/pub/phod/lumpkin/netcdf/, we are using. The latest there is `buoydata_15001_feb23/` but the datasets has a `gdp6h_ragged_may23` timestamp...

File: clouddrift/datasets.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def gdp6h(decode_times: bool = True) -> xr.Dataset:
     --------
     :func:`gdp1h`
     """
-    url = "https://www.aoml.noaa.gov/ftp/pub/phod/buoydata/gdp_jul22_ragged_6h.nc#mode=bytes"
+    url = "https://www.aoml.noaa.gov/ftp/pub/phod/buoydata/gdp6h_ragged_may23.nc#mode=bytes"
     ds = xr.open_dataset(url, decode_times=decode_times)
     ds = ds.rename_vars({"ID": "id"}).assign_coords({"id": ds.ID}).drop_vars(["ids"])
```
Comment (philippemiron) [inline clouddrift/datasets.py:149]: and I guess this should assign `ds.id` as coordinates?

**Comment by philippemiron** [comment]
Oh wait, I see that the time are not decoded when running the adapters. Sorry, I will fix later.

**Comment by philippemiron** [comment]
from clouddrift.adapters.gdp6h import to_raggedarray
ds = to_raggedarray(n_random_id=250).to_xarray()
ds

```
<xarray.Dataset>
Dimensions:                (traj: 250, obs: 404113)
Coordinates:
    ids                    (obs) int64 7704404 7704404 ... 300534062478470
    time                   (obs) float64 4.056e+08 4.056e+08 ... 1.696e+09
Dimensions without coordinates: traj, obs
Data variables: (12/50)
    ID                     (traj) int64 7704404 7706856 ... 300534062478470
    rowsize                (traj) int32 2186 1226 3465 152 ... 678 470 375 73
    WMO                    (traj) int32 0 4300502 5100512 ... 3801630 7801653
    expno                  (traj) int32 289 129 129 2425 ... 21412 21412 2222
    deploy_date            (traj) float32 4.056e+08 5.614e+08 ... 1.695e+09
    deploy_lat             (traj) float32 -0.01 10.18 0.0 ... 40.85 34.43 64.11
    ...                     ...
    vn                     (obs) float32 nan -0.08299 -0.03291 ... -0.1195 nan
    temp                   (obs) float32 26.2 26.01 26.07 ... 9.128 9.054 9.047
    err_lat                (obs) float32 0.006709 0.006949 ... 0.002629 0.001429
    err_lon                (obs) float32 0.006622 0.00706 ... 0.00327 0.00203
    err_temp               (obs) float32 0.06973 0.08061 ... 0.03661 0.03768
    drogue_status          (obs) bool True True True True ... True True True
Attributes: (12/18)
    title:                Global Drifter Program hourly drifting buoy collection
    history:              version September 2023. Metadata from dirall.dat an...
    Conventions:          CF-1.6
    time_coverage_start:  1982-11-08:06:00:00Z
    time_coverage_end:    2023-10-18:18:00:00Z
    date_created:         2023-12-21T12:51:10.882231
    ...                   ...
    contributor_name:     NOAA Global Drifter Program
    contributor_role:     Data Acquisition Center
    institution:          NOAA Atlantic Oceanographic and Meteorological Labo...
    acknowledgement:      Lumpkin, Rick; Centurioni, Luca (2019). NOAA Global...
    summary:              Global Drifter Program six-hourly data
    doi:                  10.25921/7ntx-z961
```


from clouddrift.adapters.gdp1h import to_raggedarray
ds = to_raggedarray(n_random_id=250).to_xarray()
ds


```
<xarray.Dataset>
Dimensions:                (traj: 250, obs: 2548777)
Coordinates:
    ids                    (obs) int64 9729735 9729735 ... 300234011680550
    time                   (obs) float64 8.843e+08 8.844e+08 ... 1.667e+09
Dimensions without coordinates: traj, obs
Data variables: (12/60)
    ID                     (traj) int64 9729735 9729757 ... 300234011680550
    rowsize                (traj) int32 32877 25876 5711 23462 ... 2398 1742 481
    WMO                    (traj) int32 3400917 4600606 0 ... 4602555 5102844
    expno                  (traj) int32 9325 9325 1195 ... 31412 21312 31412
    deploy_date            (traj) float32 8.842e+08 8.98e+08 ... 1.665e+09
    deploy_lat             (traj) float32 -50.0 53.0 -30.58 ... 30.73 -3.0
    ...                     ...
    err_sst1               (obs) float32 0.065 0.01 0.01 ... 0.002058 0.002033
    err_sst2               (obs) float32 0.074 0.025 0.026 ... 0.005325 0.005337
    flg_sst                (obs) int8 5 5 5 5 5 5 5 5 5 5 ... 5 5 5 5 5 5 5 5 5
    flg_sst1               (obs) int8 5 5 5 5 5 5 5 5 5 5 ... 5 5 5 5 5 5 5 5 5
    flg_sst2               (obs) int8 2 2 2 2 2 2 2 5 5 5 ... 5 5 5 5 5 5 2 5 5
    drogue_status          (obs) bool True True True True ... True True True
Attributes: (12/18)
    title:                Global Drifter Program hourly drifting buoy collection
    history:              version 2.01. Metadata from dirall.dat and deplog.dat
    Conventions:          CF-1.6
    time_coverage_start:  1998-01-09:12:00:00Z
    time_coverage_end:    2022-10-31:00:00:00Z
    date_created:         2023-12-21T12:50:30.984873
    ...                   ...
    contributor_name:     NOAA Global Drifter Program
    contributor_role:     Data Acquisition Center
    institution:          NOAA Atlantic Oceanographic and Meteorological Labo...
    acknowledgement:      Elipot, Shane; Sykulski, Adam; Lumpkin, Rick; Centu...
    summary:              Global Drifter Program hourly data
    doi:                  10.25921/x46c-3620
```

**Comment by selipot** [comment]
Tests keep failing for Windows platforms because of the URL for the 6h GDP dataset netcdf file. I am not sure how to approach this.

**Review by selipot — APPROVED** [review]

**Comment by philippemiron** [comment]
I found this thread which seems to point to a corrupted netcdf file when getting this error.. 

https://github.com/pydata/xarray/issues/4050

It doesn't make much sense since it is working on Linux and Mac..

**Comment by philippemiron** [comment]
This thread https://stackoverflow.com/questions/49317927/errno-101-netcdf-hdf-error-when-opening-netcdf-file, suggest setting:

`HDF5_USE_FILE_LOCKING=FALSE`.

**Pull request closed without merging**